documentation/gnmi/design/sonic-dash.yang

```diff
@@ -0,0 +1,490 @@
+                        }
+                    }
+
+                    leaf encap_type {
+                        type string {
+                            pattern "vxlan|nvgre"
+                        }
+                    }
+
+                    leaf vni {
+                        type  uint32 {
+                            range 0..16777215;
+                        }
+                    }
+                }
+
+            } /* end of list DASH_ROUTING_TYPE_LIST */
+        } /* end of container DASH_ROUTING_TYPE */
+
+        container DASH_ROUTE_TABLE {
+            description "DASH Route table associated to an ENI; part of APP_DB";            
+
+            list DASH_ROUTE_TABLE_LIST {
+                key "eni prefix";
+
+                leaf eni {
+                    type leafref {
+                        path /dash:sonic-dash/dash:DASH_ENI/dash:DASH_ENI_LIST/dash:name;
+                    }
+                }
+
+                leaf prefix {
+                     type union {
+                         type stypes:sonic-ip4-prefix;
+                         type stypes:sonic-ip6-prefix;
+                     }
+                }
+ 
+                leaf ip_version {
+                    type string {
+                        pattern "IPv4|IPv6"
+                    }
+                }
+
+                leaf routing_type {
+                    type leafref {
+                        path /dash:sonic-dash/dash:DASH_ROUTING_TYPE/dash:DASH_ROUTING_TYPE_LIST/dash:name;
+                    }
+                }
+
+                leaf vnet {
+                    when "((current()/../routing_type = 'vnet') or (current()/../routing_type = 'vnet-direct'))";
+                    type leafref {
+                        path /dash:sonic-dash/dash:DASH_VNET/dash:DASH_VNET_LIST/dash:name;
+                    }
+                }
+
+                leaf customer_addr {
+                    when "((current()/../routing_type = 'vnet') or (current()/../routing_type = 'vnet-direct'))";
+                    description "if routing_type is vnet-direct"
+                    type inet:ip-address;
+                }
+
+                leaf appliance {
+                    when "(current()/../routing_type = 'appliance')";
+                    type leafref {
+                        path /dash:sonic-dash/dash:DASH_APPLIANCE/dash:DASH_APPLIANCE_LIST/dash:name;
+                    }
+                }
+
+                leaf overlay_ip {
+                    when "(current()/../routing_type = 'servicetunnel')";
+                    description "Overlay ip for service tunnel"
+                    type inet:ip-address;
+                }
+
+                leaf underlay_ip {
+                    when "(current()/../routing_type = 'servicetunnel')";
+                    description "Underlay ip for service tunnel"
+                    type inet:ip-address;
+                }
+
```

Comment:
Can we have a grouping for service tunnel specific fields and use the group name here?